src/test/java/io/nats/client/ConnectTests.java

```diff
@@ -396,26 +355,17 @@ public void testAsyncConnectionWithReconnect() throws IOException, InterruptedEx
         try {
             Nats.connectAsynchronously(options, true);
 
-            // No server at this point, let it fail and try to start over
-            try {
-                Thread.sleep(1000);
-            } catch (Exception exp) {
-
-            }
+            sleep(5000); // No server at this point, let it fail and try to start over
 
             nc = handler.getConnection(); // will be disconnected, but should be there
             assertNotNull(nc);
 
             handler.prepForStatusChange(Events.RECONNECTED);
             try (NatsTestServer ts = new NatsTestServer(port, false)) {
-                handler.waitForStatusChange(5, TimeUnit.SECONDS);
-                assertTrue(Connection.Status.CONNECTED == nc.getStatus(), "Connected Status");
+                allowTimeToConnectAssertOpen(nc, handler);

```

Comment:
This macro fixed the flappers. Some didn't allow enough time for the connection. With the macro it's consistent. Only used in tests where they are not testing the connection itself, but using the connection to test something.